analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.PatternMatching.cs

```diff
@@ -301,6 +318,23 @@ public void DeclarationPattern_Var_Deconstruction_DoesNotSetConstraint()
                 .And.ContainSingle(x => x.HasConstraint(TestConstraint.First) && x.HasConstraint(ObjectConstraint.NotNull));
         }
 
+        [TestMethod]
+        public void DeclarationPattern_ExistingConstraint_DoesNothing()
```

Comment:
It would be good to have one "ObjectConstraint.Null" example too:
```cs
object value = arg switch
{
    not null => null,
    _ => Tag("Arg", arg), // Should have "Null" constraint
};
```

Reply:
That is already tested by `ValidateSetBoolConstraint_TwoStates`